### PR TITLE
Import/Export: move import and export into dedicated sections

### LINKED ISSUE #33104: Importer Navigation and New Sidebar Usability: Remove Import and Export Sections from General
This is a sub-story of: https://github.com/Automattic/wp-calypso/issues/31169
A prerequisite of: https://github.com/Automattic/wp-calypso/issues/33105
There are sub-stories for each of the goals for better QA. Either me or @shaunandrews can design review and approve all of them.

@obi2020 or @ranh I'm tagging you for copy review, but no pressure.

**Context**
I ran a usability test with a participant using one of the importers, and found that the new sidebar makes that existing issue worse, especially if the user starts from signup.

**Goal**
To avoid navigation confusion for users completing an import. We are not telling these new users where in the product they are working. If they leave the import flow, the sidebar is not indicating where they are going, and they end up lost.

**Changes needed**
- [ ] **Remove Import and Export** sections out of **Manage>Settings>General** in the new sidebar. 
- [ ] **Add Export** section to **Tools>** in the new sidebar.
- [ ] **Remove the cake nav** from Import and Export sections that is no longer needed.

---
Here's what the new Settings sidebar section should be like:
o Import and Export

---
Here is what the new Tools>Export sidebar section should look like after moving Export into it, and removing the cake nav now that it is no longer needed:
- [ ] the two buttons can be bumped down to an `is-secondary` class.
- [ ] Type size for section title is 24px and copy underneath is 14px
1440 - Tools _ Export

---
Here is what the updated Import sidebar section should look like after removing the cake nav that is no longer needed:
Sidebar Expanded

### PR DESCRIPTION
**Pull request by creativecoder**
**IMPORTANT** Do not merge until new routes have been added to server for /import and /export sections.

#### Changes proposed in this Pull Request

Addresses #33104

* Move Import and Export pages out of `/settings` and into their own top-level sections
* Add "Export" to the sidebar under "Tools"
* Add redirects so that existing links to import and export within settings resolve to the new sections
* Design updates from https://github.com/Automattic/wp-calypso/issues/33104#issuecomment-494546650
* Removes all code references of import and export being within settings section

#### Testing instructions

- Navigate to Import and Export sections from the sidebar under "Tools" and verify they load without error.
- Import and Export items in sidebar are highlighted in sidebar when active.
- Links to `/settings/import/{site}` and `/settings/export/{site}` resolve to the appropriate section
- Links to `/settings/export/guided/` and `settings/export/guided/{host}` redirect to appropriate `/export` pages.
- Links to `/settings/import?engine=wix&from_url={url}` lands you on the Wix importer with the url input filled in with `from_url` query param.
- Run through the `/start/import` signup flow and verify that you land on the Wix or GoDaddy importer with the url you entered during signup.
- Test import and export functionality still works as expected.
- Review design changes from https://github.com/Automattic/wp-calypso/issues/33104#issuecomment-494546650, including Import and Export section headers and copy, and button styles for export section.
- Go to `/settings/general/{site}` and verify that import and export sections have been removed at the bottom of the page.

## Tour of this PR

With 47 files changed, I recognize this one is time-intensive to review. Here's a comprehensive tour of all files changed in this PR.

#### Updating header for import and export sections

- [client/components/formatted-header/index.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-e6889573c66bd2f9f93a4ec9688d3eeb) - Making use of `className` prop in `FormattedHeader` component
- [client/my-sites/site-settings/settings-import/descriptive-header.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-abcac7246789b82b036f24a6dcf163b2) - removing now unused `DescriptiveHeader` component

##### Export section header

- [client/my-sites/exporter/section-export.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-d6d17a8a85a1ada006b1dbc52049f776)
- [client/my-sites/exporter/style.scss](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-5a89568eb771dcc4cb03d4cda26799ce)

##### Import section header

- [client/my-sites/importer/section-import.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-96b1bf86f4ccfd0ce24bfbe608174d91)
- [client/my-sites/importer/style.scss](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-3926fad84c71cc88785094c2d8f48692)

#### Remove primary styling for all buttons on export page

- [client/my-sites/exporter/export-card/advanced-settings.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-10a30a2a1d87bf9da090ae6dbb59834e)
- [client/my-sites/exporter/export-card/index.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-4f154ee15f1755509a0e8a9eaeb4c39d)
- [client/my-sites/exporter/export-media-card/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-14b8024f3bb40c493511da1b03b14fd7)
- [client/my-sites/exporter/guided-transfer-card/index.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-9ffa2ed031bd2842429063152f5b7467)

### 2. Creating new top-level sections for `import` and `export`

- [client/sections.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-81243a1c2b5d87c4c614a9ee048b7614)

##### Removing import/export from settings controller

- [client/my-sites/site-settings/controller.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-5ce7e1b12afe6d1a309f9f291f36e997)
- [client/my-sites/site-settings/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-f8814476d7d4b72688289e9a2971d697) - including adding redirects for `settings/import` and `settings/export` routes

##### Export controller and routes

- [client/my-sites/exporter/controller.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-fb30069e4ef579a9fcef8154ca949490)
- [client/my-sites/exporter/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-7b0ad5e0a18bd5de43d25c6ae01115a4)

##### Import controller and routes

- [client/my-sites/importer/controller.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-85eab9885a6aab8159915ee1d2897626)
- [client/my-sites/importer/index.js](client/my-sites/importer/index.js)

#### Moving guided transfer (a subsection of `export`)

- [client/my-sites/guided-transfer/guided-transfer.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-e9fe6eba2e70ac2d57d294e559aa1599)

#### Removing import and export items from settings page

- [client/my-sites/site-settings/site-tools/index.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-5eb6e236b859e37b3ed53607ba386b0c)

### 3. Sidebar changes

- [client/my-sites/sidebar/sidebar.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-3e29fa10a6f1fb8927b10336ff9ff66b)
- [client/my-sites/sidebar/tools-menu.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-63c9226a6df970fa1cd73cd0bd98c775)

### 4. Updating links and references to `/settings/import` and `/settings/export`

- [client/my-sites/site-settings/delete-site/index.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-07c88ddb1727df6930b0e317bbc383c0)
- [client/signup/config/flows-pure.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-28b03eff6ef567c591024d6033cda11d)

#### Renaming components and/or files

- [client/my-sites/exporter/container.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-aa4463ddb772e82673a68f50e503e4cc)
- [client/my-sites/importer/error-pane.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-a7ac2fec1be87a909285c9b428991035) - required updating some `a` tags to `Button` to make eslint happy
- [client/my-sites/importer/importing-pane.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-462ebecb0ee52e7ab96506e1cd573a43)
- [client/my-sites/importer/jetpack-importer.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-ad7aeb7a1b3d0147a8ea253c780570f3)

#### Moving `state/site-settings/exporter` to `state/exporter`

- [client/state/exporter/actions.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-e0bc7725067a5716a97690e50492e480)
- [client/state/exporter/constants.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-be7376f89c9917c9cb53c6b4ce0b62c4)
- [client/state/exporter/reducers.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-ea3a7e38a902374c12409db3cef2d09a)
- [client/state/exporter/selectors.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-5dcea1366778dd247d9204a5dc23d37e)
- [client/state/reducer.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-e7517e5bf5dccdc38b64758a5d0794da)
- [client/state/selectors/get-media-export-url.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-81d21ea1d7ad829fe51c439fa0d0bee6)
- [client/state/site-settings/reducer.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-1ab0bf0849bae3445266a693e69d5d36)

#### Corresponding updates to redux tests

- [client/state/exporter/test/actions.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-ee06a555559bd8321bf596a01764ebaa)
- [client/state/exporter/test/data/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-098cd45d8e2afc3a8f3293c822dad6e5)
- [client/state/exporter/test/reducer.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-80a6f53640629843c58e5522255353c7)
- [client/state/exporter/test/selectors.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-57b35a3c262ec9a9e1940e2a24694ee7)
- [client/state/selectors/test/get-media-export-url.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-9e27cfd43ea65a498aa03ab1508d26d6)

#### And updating references to `state/exporter`

- [client/components/data/query-media-export/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-f041a72ca58312a53eac2fffbc41b2fa)
- [client/my-sites/exporter/export-card/post-type-options.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-035e12770bf0bc6f6004850617ff248f)
- [client/my-sites/exporter/export-card/select.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-0a8309bd7972cdabf01faab476a656d5)
- [client/my-sites/exporter/notices.jsx](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-aac3d09eb8f66f97fe6106d6adc85c9c)
- [client/state/data-layer/wpcom/sites/exports/media/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-889e1ff5c8b9ffc955ac362cea4ce132)
- [client/state/data-layer/wpcom/sites/exports/media/test/index.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-acae45a0432fe344bdb370982ea2cb47)

### 5. Updates to e2e tests

- [test/e2e/lib/pages/importer-page.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-450f6bdfda3b14b69b05978d8ef73796)
- [test/e2e/specs/wp-import-site-spec.js](https://github.com/Automattic/wp-calypso/pull/33181/files#diff-c8da8f4b6214908db726e96acb24d9db)

### CODE REVIEW

**Comment by creativecoder** [comment]
@Automattic/team-calypso I noticed that moving import and export into separate sections has a [heavy impact on bundle size](https://github.com/Automattic/wp-calypso/pull/33181#issuecomment-494172769) for "Sections" and "Async-loaded Components". Would it be better to combine these into a single section, maybe under `/tools`? Any other advice performance-wise?

**Comment by jsnajdr** [comment]
Hi @creativecoder :wave: The first thing I would recommend doing is removing the async load of the import/export components:
```js
context.primary = <AsyncLoad require="my-sites/importer/section-import" />
context.primary = <AsyncLoad require="my-sites/exporter/section-export" />
```

That was useful when these components were part of one big "Site Settings" section, but now the async loading is done by the section loader.

Just import them synchronously:
```js
import SectionImport from 'my-sites/importer/section-import';
context.primary = <SectionImport />;
```

That should merge the `import` and `async-load-my-sites-importer-section-import` chunks into one and make further investigation easier.

> Would it be better to combine these into a single section, maybe under /tools?

That shouldn't be needed. Webpack should automatically detect that both sections share code and create a shared chunk for them. That avoids duplicating code and makes merging the sections under `/tools` unneccessary performance-wise.

**Comment by dezzie** [comment]
Thanks for doing this @creativecoder :).  
Let me know if these things were not intended to be a part of this PR, and if the checklist at the top of this is valid:

1. Looks like the Export section still has some changes needed:
Screen Shot 2019-05-28 at 11 29 03 AM

- [ ] cake nav should be gone
- [ ] the new section header and subheader copy should be on there
- [ ] buttons should be secondary, not primary

---

2. The import section looks good so far:
PR export screen from signup email verified

3. The sidebar doesn't expand to indicate the part of the sidebar that the user is yet. Should I wait until something else in place before expecting this?
PR export screen from signup

**Comment by creativecoder** [comment]
> Hi @creativecoder 👋 The first thing I would recommend doing is removing the async load of the import/export components:

Thanks @jsnajdr , that exactly the feedback I was looking for 😀

**Comment by creativecoder** [comment]
@dezzie 

> Looks like the Export section still has some changes needed:

You're right, didn't get to updating that yet--add an item to the "To Do" list in the PR description so I don't forget.

> The sidebar doesn't expand to indicate the part of the sidebar that the user is yet. Should I wait until something else in place before expecting this?

Plan to handle that separately, as it's not unique to the import and export sections. None of the sidebar accordions expand to reveal which screen you are on when the page is reloaded (unless that section was already open the last time you were in Calypso). I think the cleanest way to implement this would be to have this work with all sections.

File: client/my-sites/exporter/section-export.jsx [hunk 1]
```diff
@@ -9,17 +9,19 @@ import { localize } from 'i18n-calypso';
...
 
-const SiteSettingsExport = ( { isJetpack, site, siteSlug, translate } ) => (
+const SectionExport = ( { isJetpack, site, translate } ) => (
 	<Main>
-		<HeaderCake backHref={ '/settings/general/' + siteSlug }>
-			<h1>{ translate( 'Export' ) }</h1>
-		</HeaderCake>
+		<FormattedHeader
+			className="exporter__section-header"
+			headerText={ translate( 'Export Your Content' ) }
```
Comment (a8ci18n) [inline client/my-sites/exporter/section-export.jsx]: Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'Export your content' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by creativecoder** [comment]
Whoops... looks like there are some broken unit tests... fixing now...

**Comment by creativecoder** [comment]
Unit tests resolved. This is ready for review.

**Review by jonathansadowski — COMMENTED** [review]
Since we paired on a good portion of this, I'm not going to explicitly approve it to encourage another set of eyes (but I'm happy to if nobody else is able to review).

I double-checked that all of the section redirects work as intended, and that the sections have been moved and highlighted properly. Everything looks good to me, nice work! Should we create a separate issue to track the expansion of menus when a sub-item is selected?

**Comment by creativecoder** [comment]
I've added an outline of all the files changed to the PR description--hopefully this will help make reviewing this one a little bit easier.

**Comment by creativecoder** [comment]
> Should we create a separate issue to track the expansion of menus when a sub-item is selected?

@jonathansadowski That's directly related to https://github.com/Automattic/wp-calypso/issues/33105, though I think it would make sense to either revise that issue or create a new "generic" one that describes the issue for all sidebar sections.

**Review by dezzie — APPROVED** [review]
Looks good to me, thanks @creativecoder and @jonathansadowski

**Comment by creativecoder** [comment]
Related, I've started on "auto-expanding" the sidebar https://github.com/Automattic/wp-calypso/pull/33519

**Review by pablinos — APPROVED** [review]
Looks good to me too!

I've tested the redirects, sign up flow and a standard import. It'll be good to fix #33519, but that's not for this PR.

Do we want to make one of the buttons on the export screen primary?

**Comment by dezzie** [comment]
@pablinos Nah, let's leave them both secondary buttons. We're not trying to encourage one action over another for this one, and we're trying to exercise pink-restraint :)

**Comment by creativecoder** [comment]
@pablinos Thanks for the review!

> It'll be good to fix #33519, but that's not for this PR.

I should be able to follow-up "auto-expanding" the sidebar soon (sometime this week).

**Comment by creativecoder** [comment]
I've made one addition, which redirects to `/import` and `/export` when the site is invalid (e.g. `/import/foo` is redirect to `/import` because `foo` is not a valid site).

**Review by jonathansadowski — APPROVED** [review]
Double-checked the most recent change. All of the previous redirects still work as intended, and now an invalid site slug redirects to `/import` or `/export` respectfully.

Good catch. 👍

**Pull request merged**